docs/source/run/parameters.rst

```diff
@@ -749,3 +718,53 @@ Field diagnostics
+For particle beams, they can be used to calculate the main characterizing beam parameters (width, energy spread, emittance, etc.). Additionally, the plasma particle properties (e.g, the temperature) can be calculated.
+
+The in-situ quantities are calculated per slice and together with the weights per slice, all important quantities can be calculated.
+
+For particle beams, the following quantities are calculated per slice and stored:
+``sum(w), [x], [x^2], [y], [y^2], [z], [z^2], [ux], [ux^2], [uy], [uy^2], [uz], [uz^2], [x*ux], [y*uy], [z*uz], [ga], [ga^2], np``.
+For plasma particles, the following quantities are calculated per slice and stored:
+``sum(w), [x], [x^2], [y], [y^2], [ux], [ux^2], [uy], [uy^2], [uz], [uz^2], [ga], [ga^2], np``.
+Thereby, "[]" stands for averaging over all particles in the current slice,
+"w" stands for weight, "ux" is the normalized momentum in the x direction, "ga" is the Lorentz factor.
+Averages and totals over all slices are also provided for convenience under the
+respective ``average`` and ``total`` subcategories.
+
+Additionally, some metadata is also available:
+``time, step, n_slices, charge, mass, z_lo, z_hi, normalized_density_factor``.
+``time`` and ``step`` refers to the physical time of the simulation and step number of the
+current timestep.
+``n_slices`` equal to the number of slices in the zeta direction.
+``charge`` and ``mass`` relate to a single particle and are for example equal to the
+electron charge and mass.
+``z_lo`` and ``z_hi`` are the lower and upper bounds of the z-axis of the simulation domain
+specified in the input file and can be used to generate a z/zeta-axis for plotting.
+``normalized_density_factor`` is equal to ``dx * dy * dz`` in normalized units and 1 in
+SI units. It can be used to convert ``sum(w)``, which specifies the particle density in normalized
+units and particle weight in SI units, to the particle weight in both unit systems.
+
+The data is written to a file at ``<insitu_file_prefix>/reduced_<beam/plasma name>.<MPI rank number>.txt``.
+The in-situ diagnostics file format consists of a header part in ASCII containing a JSON object.
+When this is parsed into Python it can be converted to a NumPy structured datatype.
+The rest of the file, following immediately after the closing }, is in binary format and
+contains all of the in-situ diagnostic along with some meta data. This part can be read using the
+structured datatype of the first section.
+Use ``hipace/tools/read_insitu_diagnostics.py`` to read the files using this format. Functions to calculate the most useful properties are also provided in that file.
+
+* ``<beam name> or beams.insitu_period`` (`int`) optional (default ``-1``)
+    Period of the beam in-situ diagnostics. `-1` means no in-situ diagnostics.
+
+* ``<beam name> or beams.insitu_file_prefix`` (`string`) optional (default ``"diags/insitu"``)
+    Path of the beam in-situ output. Must not be the same as `hipace.file_prefix`.
```

Comment:
I mean must not, in a sense that it is not allowed (we assert for it, because otherwise, what happens is that both the openPMD and insitu diags keep creating and overwriting the directory and one ends up with a mess and looses half the data).